File: src/code/ts/exgcd.ts
//#region main
export function exgcd(a: number, b: number): [number, number, number] {
  if (!b) return [a, 1, 0];
  const [gcd, x, y] = exgcd(b, a % b);
  return [gcd, y, x - Math.floor(a / b) * y];
}

//#endregion main

File: src/code/py/exgcd.py
#region main
def exgcd(a: int, b: int) -> int:
    if b == 0:
        return a, 1, 0
    gcd, x, y = exgcd(b, a % b)
    return gcd, y, x - a // b * y
#endregion main
